### ISSUE TITLE
Wrong number of type arguments because Result returned instead of ::std::result::Result

### ISSUE THREAD

**Issue by Michael-F-Bryan** [colin-kiegel/rust-derive-builder#52]
I recently encountered an issue where I couldn't compile because the `build()` method tries to return a `Result<T, E>`, and because I've created my own `Result` type its signature is just `Result<T>`.

```
error[E0244]: wrong number of type arguments: expected 1, found 2
  --> src/connection.rs:64:10
   |
64 | #[derive(Builder)]
   |          ^^^^^^^ expected 1 type argument
```

Would you be able to adjust the generated code so it returns the fully qualified path `::std::result::Result`?

i.e. change this:

```rust
fn build(&self) -> Result<Channel, String> {
``` 

to this:

```rust
fn build(&self) -> ::std::result::Result<Channel, String> {
```

In the meantime, if anyone else encounters this issue there's a fairly easy fix. You just need to rename your `Result` and `Error` type.

```rust
use errors::{Error as MyError, Result as MyResult};
```

**Comment by Michael-F-Bryan** [comment]
I just did a little more digging and it looks like #45 would fix this as a side effect of having to use either `core` or `std`.

**Comment by colin-kiegel** [comment]
Thanks for your feedback. It's definitely the plan to support `core` and `std` with the next release. This would then also solve the problem with local type aliases. :-)

**Comment by colin-kiegel** [comment]
published with [v0.4.0](https://github.com/colin-kiegel/rust-derive-builder/releases/tag/v0.4.0)
